src/BlockTypes/ProductQuery.php

```diff
@@ -80,41 +83,107 @@ public function get_query_by_attributes( $query ) {
+						return $acc;
+			},
+			$common_query_values
 		);
-		$on_sale_query       = $this->get_on_sale_products_query( $variation_props );
 
-		return array_merge( $query, $common_query_values, $on_sale_query );
 	}
 
+
 	/**
 	 * Return a query for on sale products.
 	 *
+	 * @return array
+	 */
+	private function get_on_sale_products_query() {
+		return array(
+			'post_in' => wc_get_product_ids_on_sale(),
+		);
+	}
+
+	/**
+	 * Set the query vars that are used by filter blocks.
+	 *
+	 * @param array $qvars Public query vars.
+	 * @return array
+	 */
+	public function set_query_vars( $qvars ) {
+		$filter_query_args = array( PriceFilter::MIN_PRICE_QUERY_VAR, PriceFilter::MAX_PRICE_QUERY_VAR );
+		return array_merge( $qvars, $filter_query_args );
+	}
+
+	/**
+	 * Return queries that are generated by query args
+	 *
+	 * @return array
+	 */
+	private function get_queries_by_applied_filters() {
+		return array( 'price_filter' => $this->get_filter_by_price_query() );
+	}
+
+	/**
+	 * Return queries that are generated by attributes
+	 *
 	 * @param array $variation_props Dedicated attributes for the variation.
 	 * @return array
 	 */
-	private function get_on_sale_products_query( $variation_props ) {
-		if ( ! isset( $variation_props['attributes']['query']['onSale'] ) || true !== $variation_props['attributes']['query']['onSale'] ) {
-			return array();
-		}
+	private function get_queries_by_attributes( $variation_props ) {
+		$on_sale_enabled = isset( $variation_props['attributes']['query']['onSale'] ) && true === $variation_props['attributes']['query']['onSale'];
+		return array(
+			'on_sale' => ( $on_sale_enabled ? $this->get_on_sale_products_query() : array() ),
+		);
+	}
+
+	/**
+	 * Return a query that filters products by price.
+	 *
+	 * @return array
+	 */
+	private function get_filter_by_price_query() {
+		$min_price = get_query_var( PriceFilter::MIN_PRICE_QUERY_VAR );
+		$max_price = get_query_var( PriceFilter::MAX_PRICE_QUERY_VAR );
+
+		$max_price_query = empty( $max_price ) ? array() : [
+			'key'     => '_price',
+			'value'   => $max_price,
+			'compare' => '<=',
+			'type'    => 'numeric',
+		];
+
+		$min_price_query = empty( $min_price ) ? array() : [
+			'key'     => '_price',
+			'value'   => $min_price,
+			'compare' => '>=',
+			'type'    => 'numeric',
+		];
 
 		return array(
 			// Ignoring the warning of not using meta queries.
 			// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
 			'meta_query' => array(
-				'relation' => 'OR',
-				array(
-					'key'     => '_sale_price',
-					'value'   => 0,
-					'compare' => '>',
-					'type'    => 'numeric',
-				),
-				array(
-					'key'     => '_min_variation_sale_price',
-					'value'   => 0,
-					'compare' => '>',
-					'type'    => 'numeric',
-				),
+				'relation' => 'AND',
+				$max_price_query,
+				$min_price_query,
```

Comment:
Should we check that `$max_price_query` and `$min_price_query` are not an empty array, here? Otherwise we end up with queries in this form:

```
    [meta_query] => Array
        (
            [relation] => AND
            [0] => Array
                (
                )

            [1] => Array
                (
                )

        )
```

(I have no idea if that's an issue, though, so feel free to ignore this comment if you think that's ok)

Reply:
I added the check in this PR https://github.com/woocommerce/woocommerce-blocks/pull/7186/files#diff-efb947e345af5e501dfaa2980b055b3b31c308b31e8db23049b6b48fecee33d8R227-R228